### PR TITLE
Enable KeyAgreement by default on DID document creation

### PR DESCRIPTION
**Pull request by dmeijboom**
* Only when the DID document is owned by the node

Fixes https://github.com/nuts-foundation/nuts-node/issues/627

### CODE REVIEW

**Review by reinkrul — APPROVED** [review]

**Comment by reinkrul** [comment]
One minor comment is that the default isn't really a default now, but a derivative, which might not be the expected behaviour of someone creating a DID document.

File: vdr/api/v1/api.go [hunk 1]
```diff
@@ -119,6 +119,11 @@ func (a Wrapper) CreateDID(ctx echo.Context) error {
 		}
 	}
 
+	// Enable `KeyAgreement` by default only when it's a DID document owned by the node.
+	if req.KeyAgreement == nil && (req.Controllers == nil || len(*req.Controllers) == 0) {
```
Comment (woutslakhorst) [inline vdr/api/v1/api.go:123]: not true, the DID Document can be owned by an offline key. I don't see any objections in being it enabled by default for all DID Documents created

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: vdr/api/v1/api.go (hunk 1 above)
Comment (dmeijboom) [reply vdr/api/v1/api.go:123]: All right. Then let's do that

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
Ok for the code, but:

- docs/_static/vdr/v1.yaml#DIDCreateRequest doesn't reflect the updted default value
- docs/getting-started/4-connecting-crm.rst has outdated examples

**Review by woutslakhorst — APPROVED** [review]

**Comment by woutslakhorst** [comment]
@dmeijboom due to default changes a test now fails.

**Pull request merged**